scripts/firedrake-install

```diff
@@ -1184,7 +1184,12 @@ os.environ["VIRTUAL_ENV"] = virtual_env_vars["base"]
 
 os.chdir(firedrake_env)
 
-framework = fix_osx_virtualenv_python()
+try:
+    framework = fix_osx_virtualenv_python()
+except BaseException as e:
+    log.exception(e)
+    framework = False
```

Comment:
This is OK, although I wonder, depending on which step fails, whether we can end up in a broken state but still say "success"?